.github/workflows/build-and-publish.yml

```diff
@@ -7,49 +7,88 @@ on: [workflow_dispatch]
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
+  common:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: pwsh
+    steps:
+    - uses: actions/checkout@v3
+    - name: Get Changelog Entry
+      id: changelog_reader
+      uses: mindsers/changelog-reader-action@v2
+      with:
+        validation_depth: 10
+        path: ./CHANGELOG.md
+    - name: Get Version Info
+      id: read_metadata
+      run: |
+        echo "Version: ${{ steps.changelog_reader.outputs.version }}"
+        echo "Changes: ${{ steps.changelog_reader.outputs.changes }}"
+        $owner = "${{ github.repository_owner }}".ToLower()
```

Comment:
💡☕️ Idea only: Thinking out loud and sharing, shall we make this `Env Var` https://docs.github.com/en/actions/learn-github-actions/variables#defining-environment-variables-for-a-single-workflow

Reply:
In this case I probably wouldn't bother, because it's just local to that particular step, and not shared anywhere else. But that's a good point - I didn't think to consider how we might be able to improve our workflow files with env vars.